### ISSUE TITLE
As a partner I want to be able to register in the system

### ISSUE THREAD

**Issue by starkmsu** [OpenMAVN/PM#42]
Need implement REGISTRATION feature for the Small Business/PARTNER on the current login page of Admin Portal.

For this we need:
1) After SME self-registration, corresponding admin user is created with limited permissions
2) For created admin we auto-create a partner (with the name filled during the registration)
3) We need to make the following fields required for initial partner registration:
- Company name
- Email 
- Password 
- Confirm Password
4) Email verification must be introduced for self-registration

**Comment by ElenaBelyavskaya** [comment]
@starkmsu On initial registration, let's verify phone numbers instead of email? Emails are easier created/multiplied and therefore easier faked than phone numbers. Is there a reason not to use sms services?
Also I suggest we add Company Name field - so that we could create partner with this name right after the registration.

**Comment by ElenaBelyavskaya** [comment]
I propose that we:
1) Include the following fields on the partner onboarding page:
- Company name
- Phone number
- Password (+ Confirm Password)
2) Make verification via the phone number
3) Allow publishing campaigns only after the partner has filled all the other mandatory fields including scan of a document as partner ID (for the document I will investigate a bit more; ideally potential partners upload scans of some company ID mentioning a representing person + ID of this person)
(3 can be covered by another issue, I have raised #43)

**Comment by PaterSantyago** [comment]
Following tasks should be created/referenced:

1) [BE] Make parameters optional on Partner creations in AdminAPI - completed
2) [BE] FIlter by adminId for showing voucher campaigns - task for SmartVouchers service
3) [BE] Filter by adminId for partners - task for PartnerManagement service
4) [BE] Filter by adminId/partnerId for earn/burn rules - task for Campaigns service

**Comment by vitaliidasaev** [comment]
all subtasks deployed
